src/Castle.Facilities.AspNetCore/CastleWindsorRegistrationExtensions.cs

```diff
@@ -48,5 +51,18 @@ public static void UseCastleWindsor<TStartup>(this IApplicationBuilder app, IWin
 			container.Register(Component.For<T>());
 			app.Use(async (context, next) => { await container.Resolve<T>().Invoke(context, next); });
 		}
+
+		public static void ValidateConfiguration(this IWindsorContainer container)
+		{
+			ThrowIfFrameworkComponentsAreRegistered(container);
+		}
+
+		private static void ThrowIfFrameworkComponentsAreRegistered(IWindsorContainer container)
+		{
+			var handlers = container.Kernel.GetHandlers();
+			var hasAspNetCoreFrameworkRegistrations = handlers.Any(x => x.ComponentModel.Implementation.FullName.StartsWith("Microsoft.AspNetCore"));
+			if (hasAspNetCoreFrameworkRegistrations)
+				throw new Exception("Looks like you have implementations registered from 'Microsoft.AspNetCore'. Please do not do this as it could lead to torn lifestyles. Please remove the registrations.");
```

Comment:
I should probably rename the exception message to:

Looks like you have implementations registered from 'Microsoft.AspNetCore'. Please do not do this as it could lead to torn lifestyles. Please remove the registrations from Castle Windsor.